### PR TITLE
Law: velocity of rolling motion

### PR DESCRIPTION
**Pull request by alesanter**

### CODE REVIEW

File: symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+## v - velocity vector of a point on the object.
+## v_translation - velocity component responsible for translational movement of object's center
+## v_rotation - velocity component responsible for rotational movement of the point on object,
+##              see [rotational velocity formula](./linear_velocity_is_angular_velocity_cross_radius.py)
+
+# Notes
+## Suppose the object in question is a wheel of constant radius, then
+## - v_translation is a vector related to the movement of the wheel's center
+##   its magnitude is given by [linear velocity formula](../linear_velocity_from_angular_velocity_and_radius)
+##   where radius is the total radius of the wheel
```
Comment (blackyblack) [inline symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py:25]: But according to this formula, wheel's center has 0 radius, hence 0 linear velocity.

File: symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+## - v_rotation is calculated via [rotational velocity formula](./linear_velocity_is_angular_velocity_cross_radius.py)
```
Comment (blackyblack) [inline symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py:26]: It is the same formula as `linear_velocity_from_angular_velocity_and_radius` but for vectors.

File: symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py (hunk 1 above)
Comment (alesanter) [reply symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py:25]: I'll think about how to word it better later

File: symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+    add_cartesian_vectors,
+)
+
+# Description
+## Suppose a wheel is rolling smoothly along a surface, i.e. without slipping or bouncing on the
+## surface. Such motion can be viewed as a combination of translational motion of the wheel's center
+## of mass, and rotational motion of its points about the rotational axis.
+
+# Law: v = v_translation + v_rotation
+## v - velocity of a point on the wheel.
```
Comment (blackyblack) [inline symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py:16]: Velocity relative to what point or reference frame?

File: symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+## v_translation - velocity of the wheel's center of mass
```
Comment (blackyblack) [inline symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py:17]: Relative to what?

File: symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+## v_rotation - see [rotational velocity formula](./linear_velocity_is_angular_velocity_cross_radius.py)
```
Comment (blackyblack) [inline symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py:18]: Relative to what?

File: test/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocity_test.py [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+    QuantityVector,
+)
+from symplyphysics.laws.kinematic.vector import (
+    rolling_velocity_is_translational_plus_rotational_velocities as rolling_velocity_law,
+)
+
+# Description
+## A wheel is rolling smoothly, and for a point on the wheel the translational velocity
+## component amounts to (1, 2, -1) m/s, and the rotational one to (0, -1, 2) m/s.
+## Then the linear velocity of that point is (1, 1, 1) m/s at that point in time.
```
Comment (blackyblack) [inline test/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocity_test.py:18]: Also, describe reference frame.

File: symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+def rolling_velocity_law(translational_velocity_: Vector, rotational_velocity_: Vector) -> Vector:
+    return add_cartesian_vectors(translational_velocity_, rotational_velocity_)
+
+
+@validate_input(translational_velocity_=units.velocity, rotational_velocity_=units.velocity)
+@validate_output(units.velocity,)
+def calculate_rolling_velocity(
+    translational_velocity_: QuantityVector, rotational_velocity_: QuantityVector,
+) -> QuantityVector:
+    result = rolling_velocity_law(translational_velocity_, rotational_velocity_)
```
Comment (blackyblack) [inline symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py:31]: Now we have to explicitly convert to Vector. See other laws for reference.

File: symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+@validate_input(translational_velocity_=units.velocity, rotational_velocity_=units.velocity)
+@validate_output(units.velocity,)
+def calculate_rolling_velocity(
+    translational_velocity_: QuantityVector, rotational_velocity_: QuantityVector,
+) -> QuantityVector:
+    result = rolling_velocity_law(
+        translational_velocity_.to_base_vector(),
+        rotational_velocity_.to_base_vector(),
+    )
+    return QuantityVector(result.components, translational_velocity_.coordinate_system)
```
Comment (blackyblack) [inline symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py:35]: Use `QuantityVector.from_base_vector()`

**Review by blackyblack — APPROVED** [review]

File: symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py (hunk 8 above)
Comment (alesanter) [reply symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py:35]: Okay, I'll check what's new with `QuantityVector`s more thoroughly

Comment (blackyblack) [reply symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py:35]: It is the same line of code, but extracted to special function.

Comment (blackyblack) [reply symplyphysics/laws/kinematic/vector/rolling_velocity_is_translational_plus_rotational_velocities.py]: Sounds like a good idea

**Review by blackyblack — APPROVED** [review]

**Pull request merged**